BEIMA.Backend/ReportService/ReportWriter.cs

```diff
@@ -0,0 +1,260 @@
+﻿using BEIMA.Backend.Models;
+using BEIMA.Backend.MongoService;
+using MongoDB.Bson;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+
+namespace BEIMA.Backend.ReportService
+{
+    /// <summary>
+    /// This class abstracts basic report generation operations.
+    /// </summary>
+    public static class ReportWriter
+    {
+        /// <summary>
+        /// Generates a byte[] containing a file object. The
+        /// file contains a header based on the paramter deviceType
+        /// and then the data of all devices related to the specific deviceTypeId.
+        /// </summary>
+        /// <param name="deviceType"></param>
+        /// <param name="devices"></param>
+        /// <param name="delimiter"></param>
```

Comment:
Sorry to be that guy, but we needs ze param comments. Can just use "The device type", "The list of devices to enter", "the delimeter" etc.